src/dune_file.ml

```diff
@@ -1788,6 +1788,60 @@ module Menhir = struct
        })
 end
 
+module Coq = struct
+
+  type t =
+    (* ; public     : Public_lib.t option *\) *)
+    { name       : Loc.t * string
+    (* TODO: validate name *)
```

Comment:
what kind of validation would be necessary?

Reply:
Coq has certain rules in what it accepts as a library name.

Reply:
I would suggest to have an abstract type like `Coq_lib_name.t` or `Coq_lib.Name.t`. Previously we had this proliferation of string types everywhere in dune and it made the source code quite a bit harder to read.

Reply:
This is on the roadmap @rgrinberg , I can introduce `Coq_lib` [or ` Coq_library` ?] now if you want.

Reply:
This is solved in the next PR adding the install rules for the files.